### PR TITLE
[beta] Update LLVM to resolve a miscompilation found in 114312. 

### PR DESCRIPTION
**Pull request by dianqk**
Related issue: #114312 .

After the master updates the LLVM, we will add the same test cases. In the meantime, close the issue.

### CODE REVIEW

File: tests/codegen/issues/issue-114312.rs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+// compile-flags: -O
+// min-llvm-version: 16
```
Comment (cuviper) [inline tests/codegen/issues/issue-114312.rs:2]: This isn't going to work with vanilla LLVM 16, but a "system" qualifier on 17 will still let it run with our bundled 16.
```suggestion
// min-system-llvm-version: 17
```

Comment (DianQK) [reply tests/codegen/issues/issue-114312.rs]: I get it, thanks.

**Comment by cuviper** [comment]
Thanks! We're outside normal procedure, but I'm accepting this backport based on <https://github.com/rust-lang/rust/issues/114312#issuecomment-1673341072>.
@rustbot label +beta-accepted -beta-nominated
@bors r+ rollup=never

**Comment by DianQK** [comment]
:broken_heart: Ooooooops!!
https://rust.godbolt.org/z/sfePo7bTb

I'm not sure why rustc didn't submit `byval` for aarch64. For this PR I would change it to only check on x86. Also, add a byval and readonly check.

**Comment by cuviper** [comment]
> I'm not sure why rustc didn't submit `byval` for aarch64.

I would assume that's just a difference in calling convention for their C ABIs.

**Comment by DianQK** [comment]
> > I'm not sure why rustc didn't submit `byval` for aarch64.
> 
> I would assume that's just a difference in calling convention for their C ABIs.

Done. https://github.com/rust-lang/rust/compare/8ca16b2ae5701afccfe51055a0425fb3fa6127de..3008c3ee85223e69430e1e8712af4b683e33eb86. I hope this goes well.

**Comment by DianQK** [comment]
Change to `only-x86_64-unknown-linux-gnu`.

**Pull request merged**